shared/js/ui/base/chrome-ui-wrapper.es6.js

```diff
@@ -0,0 +1,61 @@
+let fetch = ((message) => {
+    return new Promise((resolve, reject) => {
+        window.chrome.runtime.sendMessage(message, (result) => resolve(result))
+    })
+})
+
+let backgroundMessage = ((thisModel) => {
+    // listen for messages from background and
+    // // notify subscribers
+    window.chrome.runtime.onMessage.addListener((req) => {
+        if (req.whitelistChanged) thisModel.send('whitelistChanged')
+        if (req.updateTabData) thisModel.send('updateTabData')
+        if (req.didResetTrackersData) thisModel.send('didResetTrackersData', req.didResetTrackersData)
+        if (req.closePopup) window.close()
+    })
+})
+
+let getBackgroundTabData = ((thisModel) => {
+    return new Promise((resolve, reject) => {
+        thisModel.fetch({getCurrentTab: true}).then((tab) => {
+            if (tab) {
+                thisModel.fetch({getTab: tab.id}).then((backgroundTabObj) => {
+                    resolve(backgroundTabObj)
+                })
+            }
+        })
+    })
```

Comment:
[nitpick] why does this need to go via `thisModel`? Seems you can just use the `fetch` that's within this module.

Reply:
👍